### PR TITLE
fix issue #42

### PR DESCRIPTION
**Pull request by LopatkinEvgeniy**
Implement method "touch"

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -192,6 +192,42 @@ module.exports = function(session) {
...
+
+		debug_log('Touching session: ' + session_id);
+
+		var expires;
+
+		if (data.cookie && data.cookie.expires) {
+			expires = data.cookie.expires;
+		} else {
+			expires = new Date(Date.now() + this.options.expiration);
+		}
```
Comment (chill117) [inline lib/index.js]: I think the touch method should always set a new value for expires. If the old value is used, the session will expire eventually even if touch is used.

File: .gitignore [hunk 2]
```diff
@@ -1 +1,2 @@
-node_modules
\ No newline at end of file
+node_modules
+.idea
```
Comment (chill117) [inline .gitignore]: Is this really needed?

**Comment by chill117** [comment]
Thanks for the PR! I think this is something that should be implemented.

I've made a couple comments on the code. Also, I think it would be a good idea if there were some tests for this new method. Thanks again!

File: lib/index.js (hunk 1 above)
Comment (LopatkinEvgeniy) [inline lib/index.js]: When "data.cookie.expires" expires, session in user's browser would be deleted. We should not store it in mysql anymore. So i did not set new value to "expires" if "data.cookie.expires" is set.

PS: Excuse my poor English )))

File: .gitignore (hunk 2 above)
Comment (LopatkinEvgeniy) [inline .gitignore]: This would be convenient for using with my IDE.

Comment (chill117) [inline .gitignore]: I think it is possible to set a global git ignore value for all the git repositories on your development machine. Maybe that would be a good solution?

File: lib/index.js (hunk 1 above)
Comment (chill117) [inline lib/index.js]: Ah, that makes sense :)

**Comment by LopatkinEvgeniy** [comment]
Thank you for advice about global .gitignore. That's cool feature.

**Comment by chill117** [comment]
No problem :)

Did you want to add tests for the new touch method? I could, but it would probably not be for a few days or a week.

**Comment by LopatkinEvgeniy** [comment]
I add some tests already.

**Comment by chill117** [comment]
Ah great :)

**Pull request merged**